### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by pandyaved98**
There are some Critical and High Priority bugs.

### CODE REVIEW

**Comment by dacbd** [comment]
Thank you, there are certainly some improvements that can be made from that scanning report. The proper domain detection for example would be a great PR if you want to take that on as well 😉. 

As far as the command injection, we might be able to do more? but I feel the nature of the tool doesn't include taking unsanitized input from systems, and if a user did put `cml` in particular `cml runner` behind a service/system that took non-predetermined input the onus would be on that system to sanitize the input. Thoughts @0x2b3bfa0 ?

**Comment by dacbd** [comment]
@pandyaved98 can you update the PR more closely reflect: https://github.com/iterative/terraform-provider-iterative/blob/master/.github/workflows/codeql-analysis.yml

**Review by dacbd — CHANGES_REQUESTED** [review]
https://github.com/iterative/cml/pull/1096#issuecomment-1185829932

**Comment by pandyaved98** [comment]
## Critical 1: Expression Injection
### .github/workflows/checkbot.yml:0
```
name: checkbot
permissions:
  id-token: write
```
We can set the input value of the expression to an intermediate environment variable. And can put a limit on the permissions of any tokens used by a workflow such as the GITHUB_TOKEN.

## Critical 2: Uncontrolled command line
### src/drivers/gitlab.js:213
```
${dockerVolumesTpl} \
        ${single ? '--max-builds 1' : ''}`;

      return spawn(command, { shell: true });

*This command depends on a server-provided value.
```
If possible, use hard-coded string literals to specify the command to run or library to load. Instead of passing the user input directly to the process or library function, examine the user input and then choose among hard-coded string literals.

If the applicable libraries or commands cannot be determined at compile time, then add code to verify that the user input string is safe before using it.

##High 1: Shell command built from environment values
### src/utils.js:17
```
var cp = require("child_process"),
  path = require("path");
function cleanupTemp() {
  let cmd = "rm -rf " + path.join(__dirname, "temp");
  cp.execSync(cmd); // BAD
}

*Can be replaced with:
var cp = require("child_process"),
  path = require("path");
function cleanupTemp() {
  let cmd = "rm",
    args = ["-rf", path.join(__dirname, "temp")];
  cp.execFileSync(cmd, args); // GOOD
}
```
If possible, use hard-coded string literals to specify the shell command to run, and provide the dynamic arguments to the shell command separately to avoid interpretation by the shell.

Alternatively, if the shell command must be constructed dynamically, then add code to ensure that special characters in environment values do not alter the shell command unexpectedly.

**Comment by 0x2b3bfa0** [comment]
> _Thoughts @0x2b3bfa0?_

None of these issues seem critical from a security standpoint; i.e. command execution is required beforehand, rendering them futile. 🤷🏼‍♂️ 

Still, it would be nice to escape properly the affected values. 👍🏼

**Comment by dacbd** [comment]
@pandyaved98 if you are willing to make the requested modifications please do so; if you don't have the time or are unable to, I'll merge this to an intermediate branch to tweak. That way your contribution is not in limbo for too long.

**Comment by 0x2b3bfa0** [comment]
🔔 @pandyaved98, feel free to ping us if you need workflow run approvals; I took some days of vacation and missed the last workflow run request. 🙈

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    #   If the Autobuild fails above, remove it and uncomment the following three lines. 
+    #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
+
+    # - run: |
+    #   echo "Run, Build Application using script"
+    #   ./location_of_script_within_repo/buildscript.sh
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
```
Comment (dacbd) [inline .github/workflows/codeql-analysis.yml:72]:
```suggestion
name: CodeQL

on:
  pull_request:
    paths-ignore:
      - '**.md'
      - assets/**
  schedule:
    - cron: '0 0 * * *' # everyday @ 0000 UTC

concurrency:
  group: ${{ github.workflow }}-${{ github.ref_name }}
  cancel-in-progress: true

jobs:
  analyze:
    name: Analyze
    runs-on: ubuntu-latest
    permissions:
      actions: read
      contents: read
      security-events: write
    steps:
      - uses: actions/checkout@v3
      - uses: github/codeql-action/init@v2
        with:
          languages: javascript
      - uses: github/codeql-action/autobuild@v2
      - uses: github/codeql-action/analyze@v2
```

**Review by dacbd — CHANGES_REQUESTED** [review]

**Review by dacbd — APPROVED** [review]
enforce style changes

**Pull request merged**